docs/source/app-dev/app-arch.rst

```diff
@@ -1,295 +1,121 @@
 
-Examples of these are automation and interactive applications that let a user or code react to complex state on the ledger (e.g. the DA Navigator tool).
+You can install any of these libraries by running ``yarn add @daml/react`` in the ``ui`` directory
+of your project. Please explore the ``create-daml-app`` example project to see the usage of these
+libraries.
 
-Which approach to take
-======================
+The interface between the frontend and the backend of your application is defined by your DAML code.
+To make your life easy when interacting with the ledger, the DAML assistant can generate
+corresponding typescript data definitions for the data types declared in DAML.
 
-For all except the simplest applications, we generally recommend the state-driven approach. State-driven applications are easier to reason about when determining correctness, so this makes design and implementation easier.
+.. code-block:: bash
 
-In practice, most applications are actually a mixture of the two styles, with one predominating. It is easier to add some event handling to a state-driven application, so it is better to start with that style.
+  daml codegen ts .daml/dist/<your-project-name.dar> -o daml-ts
 
-Structuring an application
-**************************
+This will create a typescript project under ``daml-ts`` that you can install by changing directory
+to ``daml-ts`` and running ``yarn install`` and then in the root of your project ``yarn workspaces
+run build``. To start the DAML sandbox, run the ``daml-start.sh`` script at the root of the project
+directory of ``create-daml-app``.
 
-Although applications that communicate with the ledger have many purposes, they generally have some common features, usually related to their style: event-driven or state-driven. This section describes these commonalities, and the major functions of each of these styles.
+If you choose a different Javascript based frontend framework, the packages ``@daml/ledger``,
+``@daml/types`` and the generated ``@daml2ts`` package provide you with the necessary interface code
+to connect and issue commands against your ledger.
 
-In particular, all applications need to handle the asynchronous nature of the ledger API. The most important consequence of this is that applications must be multi-threaded. This is because of the asynchronous, separate streams of commands, transaction and completion events.
+Backend
+~~~~~~~
 
-Although you can choose to do this in several ways, from bare threads (such as a Java Thread) through thread libraries, generally the most effective way of handling this is by adopting a reactive architecture, often using a library such as `RxJava <https://github.com/ReactiveX/RxJava>`__.
+The backend for your application can be any DAML ledger implementation. For applications where
+performance is not of outmost importance, we recommend using the :ref:`DAML JSON API <json-api>` as
```

Comment:
I would leave out the statement about performance here. I think we don’t really have the numbers to back up that the JSON API gives you a significant performance hit and it can even be an improvement due to caching and the fact that it handles queries smarter than you might handle them yourself.